pkger/parser_test.go

```diff
@@ -1192,6 +1192,227 @@ spec:
 		})
 	})
 
+	t.Run("pkg with single dashboard xy chart", func(t *testing.T) {
+		t.Run("xy chart", func(t *testing.T) {
+			testfileRunner(t, "testdata/dashboard_xy", func(t *testing.T, pkg *Pkg) {
+				sum := pkg.Summary()
+				require.Len(t, sum.Dashboards, 1)
+
+				actual := sum.Dashboards[0]
+				assert.Equal(t, "dash_1", actual.Name)
+				assert.Equal(t, "desc1", actual.Description)
+
+				require.Len(t, actual.Charts, 1)
+				actualChart := actual.Charts[0]
+				assert.Equal(t, ChartKindXY, actualChart.Kind)
+				assert.Equal(t, 3, actualChart.Height)
+				assert.Equal(t, 6, actualChart.Width)
+				assert.Equal(t, 1, actualChart.XPosition)
+				assert.Equal(t, 2, actualChart.YPosition)
+
+				props, ok := actualChart.Properties.(influxdb.XYViewProperties)
+				require.True(t, ok)
+				assert.Equal(t, "xy", props.GetType())
+				assert.Equal(t, true, props.ShadeBelow)
+				assert.Equal(t, "xy chart note", props.Note)
+				assert.True(t, props.ShowNoteWhenEmpty)
+
+				require.Len(t, props.Queries, 1)
+				q := props.Queries[0]
+				queryText := `from(bucket: v.bucket)  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)  |> filter(fn: (r) => r._measurement == "boltdb_writes_total")  |> filter(fn: (r) => r._field == "counter")`
+				assert.Equal(t, queryText, q.Text)
+				assert.Equal(t, "advanced", q.EditMode)
+
+				require.Len(t, props.ViewColors, 1)
+				c := props.ViewColors[0]
+				assert.NotZero(t, c.ID)
+				assert.Equal(t, "laser", c.Name)
+				assert.Equal(t, "scale", c.Type)
+				assert.Equal(t, "#8F8AF4", c.Hex)
+				assert.Equal(t, 3.0, c.Value)
+			})
+		})
+
+		t.Run("handles invalid config", func(t *testing.T) {
+			tests := []struct {
+				name      string
+				jsonStr   string
+				numErrs   int
+				errFields []string
+			}{
+				{
+					name:      "color missing hex value",
```

Comment:
lot of missing test cases here it looks like, don't think its a deal breaker here, but lets be cognizant of this moving forward. probably behooves us to put together some helpers to validate these validation errors across multiple view types

Reply:
totally agree. feels like we should do that sooner than later. the in-lined json/yml is making this file harder to scroll through, and likely will get harder w/ the next set of charts